libp2p/src/main/java/io/libp2p/core/dsl/HostBuilder.java

```diff
@@ -72,8 +79,20 @@ public Host build() {
     return BuilderJKt.hostJ(
         defaultMode_.asBuilderDefault(),
         b -> {
-          b.getIdentity().random();
+          IdentityBuilder identity = b.getIdentity();
+          identity.random(KeyType.ED25519);
+          PrivKey peerId = identity.getFactory().invoke();
+          identity.setFactory(() -> peerId);
 
+          secureTransports_.forEach(
+              st ->
+                  b.getSecureTransports()
+                      .add(
+                          p ->
+                              u ->
+                                  st.apply(
+                                      identity.getFactory().invoke(),
+                                      (List<ProtocolBinding<?>>) p)));
```

Comment:
Not sure I understood what is this code for ?

Reply:
This is to convert the secure transport constructors into normal transport constructors, after passing in the mutable protocols list.

Reply:
p is the protocol list, u is the connection upgrader (not used), st is the secure transport constructor

Reply:
Could you please check this PR https://github.com/Peergos/jvm-libp2p/pull/14/files - I've tried to untangle this transport ctor mess a bit